### PR TITLE
Add integration to all targets

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Adds a codegen test to each host platform (I recently discovered that it didn't work on mac). This also fixes the issue where we attempt to initialize-from-dill with a file that used a different multi-root scheme. Instead pass an non-existent file when using codegen, at least until I track down the underlying cause in fe server.

## Related Issues

https://github.com/flutter/flutter/issues/27141

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes tests for *all* changed/updated/fixed behaviors (See [Test Coverage]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/build_runner/build_runner.dart [hunk 1]
```diff
@@ -92,6 +87,8 @@ class BuildRunner extends CodeGenerator {
         }
       }
       stringBuffer.writeln('  build_runner: ^$kMinimumBuildRunnerVersion');
+      stringBuffer.writeln('dependency_overrides:');
+      stringBuffer.writeln('    analyzer: 0.35.3');
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/build_runner/build_runner.dart]: Maybe make the hard-coded analyzer version into a named constant next to `kMinimumBuildRunnerVersion`?

File: packages/flutter_tools/lib/src/codegen.dart [hunk 2]
```diff
@@ -179,10 +184,10 @@ class CodeGeneratingResidentCompiler implements ResidentCompiler {
         fs.path.join(flutterProject.generated.path, 'lib${platform.pathSeparator}'),
         fs.path.join(flutterProject.directory.path, 'lib${platform.pathSeparator}'),
       ],
-      fileSystemScheme: _kMultiRootScheme,
+      fileSystemScheme: kMultiRootScheme,
       targetModel: TargetModel.flutter,
       unsafePackageSerialization: unsafePackageSerialization,
-      initializeFromDill: initializeFromDill,
+      initializeFromDill: ' ',
```
Comment (yjbanov) [inline packages/flutter_tools/lib/src/codegen.dart]: What does the magic value `' '` do? Can we document it somewhere?

**Review by yjbanov — APPROVED** [review]
lgtm

**Comment by jonahwilliams** [comment]
Tests passed but status didn't update, landing!

**Pull request merged**